liquidctl/driver/nvidia.py

```diff
@@ -0,0 +1,171 @@
+            return
+
+        supported = [
+            (_NVIDIA_GTX_1080, GTX_1080_FTW, "EVGA GTX 1080 FTW (experimental)"),
+        ]
+
+        for (dev_id, sub_dev_id, desc) in supported:
+            if (product and product != sub_dev_id) \
+                    or (match and match.lower() not in desc.lower()) \
+                    or smbus.parent_subsystem_device != sub_dev_id \
+                    or smbus.parent_device != dev_id \
+                    or not smbus.description.startswith('NVIDIA i2c adapter 1 '):
+                continue
+
+            dev = cls(smbus, desc, vendor_id=_EVGA, product_id=GTX_1080_FTW,
+                      address=cls.ADDRESS)
+            _LOGGER.debug('instanced driver for %s', desc)
+            yield dev
+
+    def get_status(self, verbose=False, **kwargs):
+        """Get a status report.
+
+        Returns a list of `(property, value, unit)` tuples.
+        """
+
+        # only RGB lighting information can be fetched for now; as that isn't
+        # super interesting, only enable it in verbose mode
+
+        if not verbose:
+            return []
+
+        if not check_unsafe('smbus', 'evga_pascal', **kwargs):
+            _LOGGER.warning("%s: nothing to return, requires unsafe features "
+                            "'smbus,evga_pascal'",  self.description)
+            return []
+
+        mode = self.Mode(self._smbus.read_byte_data(self.ADDRESS, self.REG_MODE))
+        status = [('Mode', mode, '')]
+
+        if mode.required_colors > 0:
+            r = self._smbus.read_byte_data(self.ADDRESS, self.REG_RED)
+            g = self._smbus.read_byte_data(self.ADDRESS, self.REG_GREEN)
+            b = self._smbus.read_byte_data(self.ADDRESS, self.REG_BLUE)
+            status.append(('Color', f'{r:02x}{g:02x}{b:02x}', ''))
+
+        return status
+
+    def set_color(self, channel, mode, colors, non_volatile=False, **kwargs):
+        """Set the RGB lighting mode and, when applicable, color.
+
+        The table bellow summarizes the available channels, modes and their
+        associated number of required colors.
+
+        | Channel  | Mode      | Required colors |
+        | -------- | --------- | --------------- |
+        | led      | off       |               0 |
+        | led      | fixed     |               1 |
+        | led      | breathing |               1 |
+        | led      | rainbow   |               0 |
+
+        The settings configured on the device are normally volatile, and are
+        cleared whenever the graphics card is powered down.
+
+        It is possible to store them in non-volatile controller memory by
+        passing `non_volatile=True`.  But as this memory has some unknown yet
+        limited maximum number of write cycles, volatile settings are
+        preferable, if the use case allows for them.
+        """
+
+        check_unsafe('smbus', 'evga_pascal', error=True, **kwargs)
+
+        channel = 'led'
```

Comment:
should the channel name that is given be checked against the value 'led'? Right now channel can be set to anything and is ignored.

Reply:
We could emit a warning here, but since it's unlikely to ever become a hard error (we're not going to find/expose new LED channels), I don't think it's necessary.

The reason why it wouldn't be a hard error is that I tend to prefer automatically correcting user input when that's possible/safe.  It's the same idea with when the user supplies too many colors.

Reply:
That makes sense.